pymilvus/milvus_client/milvus_client.py

```diff
@@ -0,0 +1,801 @@
+            timeout (int, optional): The timeout to use, will override init timeout. Defaults
+                to None.
+            batch_size (int, optional): The batch size to perform inputs with. Defaults to 100.
+            partition (str, optional): Which partition to insert into. Defaults to None.
+            progress_bar (bool, optional): Whether to display a progress bar for the input.
+                Defaults to False.
+        Returns:
+            List[Union[str, int]]: A list of primary keys that were inserted.
+        """
+        # If the collection exists we need to first delete the values
+        if self.collection is not None:
+            pks = [x[self.pk_field] for x in data]
+            self.delete_by_pk(pks, timeout)
+
+        ret = self.insert_data(
+            data=data,
+            timeout=timeout,
+            batch_size=batch_size,
+            partition=partition,
+            progress_bar=progress_bar
+        )
+
+        return ret
+
+
+    def search_data(
+        self,
+        data: Union[List[list], list],
+        search_params: dict = None,
+        filter_expression: str = None,
+        top_k: int = 10,
+        partitions: List[str] = None,
+        timeout: int = None,
+    ) -> List[dict]:
+        """Search for a query vector/vectors.
+
+        In order for the search to process, a collection needs to have been either provided
+        at init or data needs to have been inserted.
+
+        Args:
+            data (Union[List[list], list]): The vector/vectors to search.
+            search_params (dict, optional): The search params to use for the search. Will default
+                to the default set for the client.
+            filter_expression (str, optional): A filter to use for the search. Defaults to None.
+            top_k (int, optional): How many results to return per search. Defaults to 10.
+            partitions (List[str], optional): Which partitions to search within. Defaults to
+                searching through all.
+            timeout (int, optional): Timeout to use, overides the client level assigned at init.
+                Defaults to None.
+
+        Raises:
+            ValueError: The collection being searched doesnt exist. Need to insert data first.
+
+        Returns:
+            List[dict]: A list of dicts containing the score and the result data. Embeddings are
+                not included in the result data.
+        """
+
+        # TODO: Figure out thread safety
+        # with self.concurrent_lock:
+        #     self.concurrent_counter += 1
+
+        if self.collection is None:
+            logger.error("Collection does not exist: %s", self.collection_name)
+            raise ValueError(
+                "Missing collection. Make sure data inserted or intialized on existing collection."
+            )
+
+        if not isinstance(data[0], list):
+            data = [data]
+
+        return_fields = list(self.fields.keys())
```

Comment:
quick question: can the `return_fields` be choosen by the users? If the users didn't specify it, then we return all fields by default.

Reply:
Agreed, by default should be only score and id, but user should be able to specify it

Reply:
I think by default it might be better to do all fields. If a user connects to an instance and doesnt know the fields, they might want to grab all of them and parse. If they want to optimize they can say which specific ones.